src/lobster/model/_ume.py

```diff
@@ -0,0 +1,142 @@
+from typing import Callable, Literal
+
+import lightning as L
+import torch
+from torch import Tensor
+
+from lobster.constants import Modality
+from lobster.model.modern_bert import FlexBERT
+from lobster.tokenization import (
+    UmeAminoAcidTokenizerFast,
+    UmeLatentGenerator3DCoordTokenizerFast,
+    UmeNucleotideTokenizerFast,
+    UmeSmilesTokenizerFast,
+)
+
+ModalityType = Literal["SMILES", "amino_acid", "nucleotide", "3d_coordinates"]
+
+
+class Ume(L.LightningModule):
+    """Universal Molecular Encoder.
+
+    A light wrapper around FlexBert model with useful high-level functions
+    for molecular encoding across different modalities.
+
+    Parameters
+    ----------
+    checkpoint : str
+        Path to model checkpoint file.
+    freeze : bool, default=True
+        Whether to freeze the model parameters.
+
+    Examples
+    --------
+    >>> # Initialize with checkpoint
+    >>> encoder = Ume("path/to/checkpoint.ckpt")
+    >>>
+    >>> # Get embeddings for protein sequences
+    >>> sequences = ["MKTVRQERLKSIVRILERSKEPVSGAQLAEELSVSRQVIVQDIAYLRSLGYNIVATPRGYVLAGG"]
+    >>> embeddings = encoder.get_embeddings(sequences, "amino_acid")
+    >>> print(embeddings.shape)
+    torch.Size([1, 768])
+    """
+
+    def __init__(self, checkpoint: str, freeze: bool = True) -> None:
+        super().__init__()
+
+        self.freeze = freeze
+        self.model = FlexBERT.load_from_checkpoint(checkpoint)
+
+        # Freeze the model parameters
+        if self.freeze:
+            for param in self.model.model.parameters():
+                param.requires_grad = False
+
+        # Initialize tokenizers for all modalities
+        self.tokenizers = {
+            Modality.AMINO_ACID: UmeAminoAcidTokenizerFast(),
+            Modality.NUCLEOTIDE: UmeNucleotideTokenizerFast(),
+            Modality.SMILES: UmeSmilesTokenizerFast(),
+            Modality.COORDINATES_3D: UmeLatentGenerator3DCoordTokenizerFast(),
+        }
+
+    def get_tokenizer(self, inputs: list[str], modality: ModalityType) -> Callable:
+        """Get the appropriate tokenizer for the given modality.
+
+        Parameters
+        ----------
+        inputs : list[str]
+            List of input strings to encode.
+        modality : Literal["SMILES", "amino_acid", "nucleotide", "3d_coordinates"]
+            The modality to use for encoding.
+
+        Returns
+        -------
+        Callable
+            The appropriate tokenizer for the specified modality.
+
+        Examples
+        --------
+        >>> encoder = Ume("path/to/checkpoint.ckpt")
+        >>> sequences = ["MKTVRQERLKSIVRILERSKEPVSGAQL"]
+        >>> tokenizer = encoder.get_tokenizer(sequences, "amino_acid")
+        >>> tokens = tokenizer(sequences, return_tensors="pt")
+        """
+        modality_enum = Modality(modality)
```

Comment:
TODOs: this MR or future MR
can we add this as a `@property` to print the available modalities?

as well as convenience methods for saving the model and tokenizer locally and to HF, and naturalness (pseudo likelihood)